modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  BaseKey,
+  BaseTransaction,
+  InvalidTransactionError,
+  ParseTransactionError,
+  PublicKey as BasePublicKey,
+  Signature,
+  TransactionRecipient,
+  TransactionType,
+} from '@bitgo/sdk-core';
+import { AtomTransaction, TransactionExplanation, TxData } from './iface';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { Buffer } from 'buffer';
+import { AtomTransactionType, sendMsgType } from './constants';
+import utils from './utils';
+import { encodePubkey, makeSignBytes } from '@cosmjs/proto-signing';
+import { encodeSecp256k1Pubkey } from '@cosmjs/amino';
+
+export class Transaction extends BaseTransaction {
+  private _atomTransaction: AtomTransaction;
+  private _signature: Signature;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  get atomTransaction(): AtomTransaction {
+    return this._atomTransaction;
+  }
+
+  setAtomTransaction(tx: AtomTransaction): void {
```

Comment:
Based on the getter method, it seems like `set atomTransaction(...)` would be better in terms of consistency. Not a blocking change though, can be addressed in a follow up.